filebeat/crawler/prospector.go

```diff
@@ -18,6 +18,8 @@ type Prospector struct {
 	harvesterChan    chan *input.FileEvent
 	registrar        *Registrar
 	done             chan struct{}
+	harvesterStates  []input.FileState
+	stateMutex       sync.Mutex
```

Comment:
why move state back from ProspectorLog to Prospector? Generates some global coupling State and type of state. State not required for stdin, right?